docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,162 @@
+   but we try to keep the core as minimal as possible in terms of dependencies.
+* Ensure that there is a consensus about the suggested change.
+** Not all changes are discussed before they submitted as pull requests.
+   Developer mailing lists, Jira issues and JEPs are used for discussions,
+   but sometimes the changes go straight to the pull requests.
+   And we are fine with that, especially for small patches.
+** Pull requests often become a venue to discuss feasibility, underlying technical decisions and design.
+   We are fine with that as well
+** If there is no consensus about the feasibility and implementation,
+   code reviewers are expected to suggest proper channels to maintainers
+* Ensure that the pull request matches the code quality practices: test automation, Javadoc for newly introduced APIs, etc.
+** There is no goal to make the code perfect tho
+* Ensure that the pull requests contain user-friendly changelogs and, if needed, upgrade guidelines
+* Last but not least,
+  guide newcomer contributors who are not familiar with the project processes.
+
+=== Code review non-goals
+
+* Enforce a particular coding style
+** Jenkins core has a complex codebase created by many contributors,
+   and different files have different designs
+** Main goal is 
+* Make contributors to fix unrelated issues
+** Create follow-up issues instead, it is fine to referene them in comments so that the contributor might want to pick them up
+* Breakdown the commit history or squashing PRs
+** Not every contributor is a Git expert, do not request changes in the commit history unless it is necessary
+** Core maintainers can squash PRs during the merge.
+   If you feel this is important, use the `squash-me-please` label
+** We want to keep pull requests atomic when possible (one feature / fix - separate commit),
+   but we can live without it if there is no need to backport changes to the stable baseline.
+
+== Merge process
+
+=== Common merge process
+
+**Step 1: Maintainer checklist**. Merge process can be initiated once a pull request matches the requirements:
+
+* Pull request is compliant wuth requirements to submitters (see the link:/.github/PULL_REQUEST_TEMPLATE.md[pull request template])
+* There are at least 2 approvals for the pull request and no outstanding requests for change
+* Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
+* Changelog entries in the PR title and/or `Proposed changelog entries` are correct
+* Proper changelog labels are set so that the changelog can be generated automatically
+* If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set
+  and there is a `Proposed upgrade guidelines` section in the PR title.
+  (link:https://github.com/jenkinsci/jenkins/pull/4387[example])
+* It would make sense to backport the change to LTS, a Jira issue should exist and be labeled as `lts-candidate`
```

Comment:
Same as above

Reply:
@oleg-nenashev this should be as above stating that a jira should exist for backporting changes
```suggestion
* If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).
```